lib/engine.js

```diff
@@ -415,9 +415,9 @@ module.exports = function (s, conf) {
               fee = s.exchange.takerFee
             }
             trade_balance = n(s.balance.currency).divide(100).multiply(buy_pct)
-            tradeable_balance = n(s.balance.currency).divide(100 + fee).multiply(buy_pct)
+            tradeable_balance = n(s.balance.currency).divide(100 + (so.use_fee_asset ? 0 : fee)).multiply(buy_pct)
```

Comment:
Why not simply set `fee` to zero instead of mangling to code like this?

e.g.

```javascript
if (so.use_fee_asset) {
	fee = 0
} else {
	if (so.order_type === 'maker' && (buy_pct + s.exchange.takerFee < 100 || !s.exchange.makerBuy100Workaround)) {
		fee = s.exchange.makerFee
	} else {
		fee = s.exchange.takerFee
	}
}
```

Reply:
Pay fees with commission asset is not free.
Just deduct same value of commission asset instead of currency.
'fee' variable can be used for calculating other purpose.
And also guess how much deduct commission asset by log.

Reply:
> Pay fees with commission asset is not free.

If you set it 0 it has no impact on the sizing of an order, and this is what this commit is about it seems.

> 'fee' variable can be used for calculating other purpose.

No, the `fee` within executeSignal() is only used withing executeSignal() and we absolutely do want it to be persistent within the function.
Other parts will source the variable themselves as needed so this has no impact on code outside this scope.